incremental-font-transfer/src/patch_group.rs

```diff
@@ -0,0 +1,1253 @@
+use read_fonts::{tables::ift::CompatibilityId, FontRef, ReadError, TableProvider};
+use std::collections::{BTreeMap, HashMap};
+
+use crate::{
+    font_patch::{IncrementalFontPatchBase, PatchingError},
+    patchmap::{intersecting_patches, IftTableTag, PatchEncoding, PatchUri, SubsetDefinition},
+};
+
+/// A group of patches derived from a single IFT font which can be applied simultaneously
+/// to that font. Patches are initially missing data which must be fetched and supplied to
+/// the group before it can be applied to the font.
+pub struct PatchGroup<'a> {
+    font: FontRef<'a>,
+    patches: Option<CompatibleGroup>,
+}
+
+impl<'a> PatchGroup<'a> {
+    /// Intersect the available and unapplied patches in ift_font against subset_definition and return a group
+    /// of patches which would be applied next.
+    pub fn select_next_patches<'b>(
```

Comment:
is this `'b` necessary? afaict the lifetimes are the same...

Reply:
It appears so, I'm unable to get it to compile without.